src/main/java/com/aws/greengrass/mqtt/bridge/BridgeConfig.java

```diff
@@ -116,6 +145,61 @@ private static String getClientId(Topics configurationTopics) {
         return Coerce.toString(configurationTopics.findOrDefault(DEFAULT_CLIENT_ID, KEY_CLIENT_ID));
     }
 
+    private static long getAckTimeoutSeconds(Topics configurationTopics) {
+        long ackTimeoutSeconds = Coerce.toLong(configurationTopics.findOrDefault(DEFAULT_ACK_TIMEOUT_SECONDS,
+                KEY_BROKER_CLIENT, KEY_ACK_TIMEOUT_SECONDS));
+        if (ackTimeoutSeconds < MIN_TIMEOUT) {
+            LOGGER.atWarn().kv(KEY_ACK_TIMEOUT_SECONDS, ackTimeoutSeconds)
+                    .log(INVALID_CONFIG_LOG_FORMAT_STRING, KEY_ACK_TIMEOUT_SECONDS, DEFAULT_ACK_TIMEOUT_SECONDS);
+            return DEFAULT_ACK_TIMEOUT_SECONDS;
+        }
+        return ackTimeoutSeconds;
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 149:156, 160:167, 171:178, 182:189, 193:200.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.